src/words/tweet_tokenizer.jl

```diff
@@ -0,0 +1,938 @@
+
+    ts.idx = i
+    return true
+end
+
+
+"""
+    reduce_all_repeated(ts::TokenBuffer)
+
+For handling repeated characters like "helloooooo" -> :hellooo".
+"""
+function reduce_all_repeated(ts)
+    ts.idx + 4 > length(ts.input) && return false
+
+    (ts[ts.idx] == '\n' || ts[ts.idx] != ts[ts.idx + 1] ||
+              ts[ts.idx] != ts[ts.idx + 2]) && return false
+
+    i = ts.idx + 3
+    while i <= length(ts.input) && ts[i] == ts[ts.idx]
+        i += 1
+    end
+    for j in 1:3
+        push!(ts.buffer, ts[ts.idx])
+    end
+    ts.idx = i
+    return true
+end
+
+"""
+    safe_text(ts::TokenBuffer)
+
+This feature covers up for the characters where the main tokenizing function lacks
+For example - "........" -> "..." and this is detected by the key tokenizer as a
+single token of "..."
+"""
+function safe_text(ts)
+    ts.idx + 4 > length(ts.input) && return false
+
+    (isalnum(ts[ts.idx]) || ts[ts.idx] != ts[ts.idx + 1] ||
+              ts[ts.idx] != ts[ts.idx + 2] )  && return false
+
+    i = ts.idx + 3
+
+    while i <= length(ts.input) && ts[i] == ts[ts.idx]
+        i += 1
+    end
+
+    for j in 1:3
+        push!(ts.buffer, ts[ts.idx])
+    end
+    ts.idx = i
+
+    return true
+end
+
+
+"""
+    replace_html_entities(input::AbstractString, remove_illegal=true)
+
+
+`input::AbstractString` The string on which HTML entities need to be replaced
+`remove_illegal::Bool` If `true`, entities that can't be converted are
+removed. Otherwise, entities that can't be converted are kept "as
+is".
+
+"""
+function replace_html_entities(input::AbstractString; remove_illegal=true)
+    ts = TokenBuffer(input)
+    isempty(input) && return ""
+
+    while !isdone(ts)
+        html_entity(ts, remove_illegal) || character(ts)
+    end
+    return ts.tokens[1]
+end
+
+
+"""
+    function  pre_process(input::AbstractString, strip_handle::Bool,
+                            reduce_len::Bool)
+
+This function processes on the input string and optionally remove twitter handles
+and reduce length of repeated characters (like "waaaaay" -> "waaay")
+and for elements like ".........?????? -> "...???" to increase the performance
+of the key tokenizer.
+"""
+function pre_process(input::AbstractString, strip_handle::Bool, reduce_len::Bool)
+    ts = TokenBuffer(input)
+    isempty(input) && return ""
+
+    while !isdone(ts)
+        (strip_handle && strip_twitter_handle(ts)) ||   # Remove username handles
+        (reduce_len && reduce_all_repeated(ts)) ||    # Reduce Lengthening
+        safe_text(ts) ||  # Shorten some sequences of characters
+        character(ts)
+    end
+
+    return ts.tokens[1]
+end
+
+function flushaboutindex!(ts::TokenBuffer, uptoidx)
+    flush!(ts, String(ts[ts.idx:uptoidx]))
+    ts.idx = uptoidx + 1
+    return true
+end
+
+const forehead = ['>', '<']
+const eyes = [':' ';' '=' '8']
+const nose = ['-','o','*','\'']
+const mouth = [')', ']', '}', '(', '[', '{', 'd', 'D', 'p', 'P', '\\', '/', ':', '@', '|']
```

Comment:
```suggestion
const mouth = (')', ']', '}', '(', '[', '{', 'd', 'D', 'p', 'P', '\\', '/', ':', '@', '|')
```

Reply:
Similar for all other constant vectors like this.
Tuples are immutable and faster